python/lsst/pipe/base/pipelineIR.py

```diff
@@ -902,17 +897,20 @@ def to_file(self, filename: str):
         """
         self.write_to_uri(filename)
 
-    def write_to_uri(self, uri: Union[ButlerURI, str]):
+    def write_to_uri(
+        self,
+        uri: ResourcePathExpression,
+    ):
         """Serialize this `PipelineIR` object into a yaml formatted string and
         write the output to a file at the specified uri.
 
         Parameters
         ----------
-        uri: `str` or `ButlerURI`
+        uri: convertible to `ResourcePath`
             Location of document to write a `PipelineIR` object.
         """
-        butlerUri = ButlerURI(uri)
-        butlerUri.write(yaml.dump(self.to_primitives(), sort_keys=False).encode())
+        with ResourcePath(uri).open("w") as buffer:
```

Comment:
@TallJimbo I was torn whether to keep the I/O code as it was or adopt the new form that you had written on the branch (also the `from_uri` method above).

Reply:
No opinion from me.